python/tests/unit/contents/test_chat_history.py

```diff
@@ -315,6 +317,30 @@ async def test_template(chat_history: ChatHistory):
     assert chat_history_2.messages[2].role == AuthorRole.USER
 
 
+@pytest.mark.asyncio
+async def test_template_safe(chat_history: ChatHistory):
+    chat_history.add_assistant_message("I am an AI assistant")
+
+    template = "system stuff{{$chat_history}}{{$input}}"
+    rendered = await KernelPromptTemplate(
+        prompt_template_config=PromptTemplateConfig(name="test", description="test", template=template)
+    ).render(
+        kernel=Kernel(),
+        arguments=KernelArguments(chat_history=chat_history, input="What can you do?"),
+    )
+    assert "system stuff" in rendered
+    assert "I am an AI assistant" in rendered
+    assert r"What%20can%20you%20do%3F" in rendered
+
+    chat_history_2 = ChatHistory.from_rendered_prompt(rendered)
+    assert chat_history_2.messages[0].content == "system stuff"
+    assert chat_history_2.messages[0].role == AuthorRole.SYSTEM
```

Comment:
Why does "system stuff" get added as a system message?

I would have expected the template would look something like:

```
<message role="system">system stuff</message>
...
<message role="user">{{$input}}</message>
```

Reply:
in Python, we allow a mix of xml and non-xml and we parse everything back into a single ChatHistory, where anything in the beginning of the template is parsed into a system message, and anything found in between or at the end into a user message, unless there is only a single message then that is turned into a user message.